filebeat/harvester/reader/multiline_test.go

```diff
@@ -131,6 +131,24 @@ func TestMultilineBeforeNegateOKWithEmptyLine(t *testing.T) {
 	)
 }
 
+func TestMultilineRepeatingFlushPattern(t *testing.T) {
+	flushMatcher := match.MustCompile(`EventEnd`)
+	pattern := match.MustCompile(`EventStart`)
+
+	testMultilineOK(t,
+		MultilineConfig{
+			Pattern:      &pattern,
+			Negate:       true,
+			Match:        "after",
+			FlushPattern: &flushMatcher,
+			LastMatch:    true,
+		},
+		2, //first two non-matching lines, will be merged to one event
+		"EventStart\nEventId: 1\nEventEnd\nEventEnd\nEventEnd\nEventEnd\n",
+		"EventStart\nEventId: 2\nEventEnd\nEventEnd\nEventEnd\n",
+	)
+}
```

Comment:
This could pass with existing options, like this:
```
func TestMultilineRepeatingFlushPattern(t *testing.T) {
        pattern := match.MustCompile(`EventStart`)

        testMultilineOK(t,
                MultilineConfig{
                        Pattern: &pattern,
                        Negate:  true,
                        Match:   "after",
                },
                2, //first two non-matching lines, will be merged to one event
                "EventStart\nEventId: 1\nEventEnd\nEventEnd\nEventEnd\nEventEnd\n",
                "EventStart\nEventId: 2\nEventEnd\nEventEnd\nEventEnd\n",
        )
}
```